utils/readers.ts

```diff
@@ -0,0 +1,30 @@
+import * as RTE from "fp-ts/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { AssertionRef } from "../generated/definitions/internal/AssertionRef";
+import {
+  LolliPOPKeysModel,
+  RetrievedLolliPopPubKeys
+} from "../model/lollipop_keys";
+import { cosmosErrorsToString, DomainError, ErrorKind } from "./errors";
+
+export type PopDocumentReader = RTE.ReaderTaskEither<
+  AssertionRef,
+  DomainError,
+  RetrievedLolliPopPubKeys
+>;
+
+// IMPLEMENTATIONS
+export const getPopDocumentReader = (
+  lollipopKeysModel: LolliPOPKeysModel
```

Comment:
Can we choose a name? lollipop, LolliPOP, lolliPOP or lolliPop? IMHO we should just use Lollipop as a single word, or just remove "lolli" and use just "pop". Lollipop is the "product name" and it is creating some naming coherence trouble in the whole codebase imo.

Reply:
if the team is concerned, I'm totally okay with unifying the name for variables and methods